### PR TITLE
Fix SILCombine of pointer_to_address instruction

### PR DESCRIPTION
**Pull request by meg-gupta**
This change ensures the insertion point of the unchecked_addr_cast is after any copies/borrows that OSSA rauw may create.

Before this fix we could end up creating invalid SIL where operands do not dominate its instruction.

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci smoke test

File: test/SILOptimizer/sil_combine_ossa.sil [hunk 1]
```diff
@@ -505,20 +505,42 @@ bb0(%0 : $*Builtin.Word, %1 : $Builtin.RawPointer):
...
   %4 = load [trivial] %3 : $*Int32
   return %4 : $Int32
 }
 
+sil @useInt : $@convention(thin) (Int32) -> ()
+
+// CHECK-LABEL: sil [ossa] @a2p_p2a_reinterpret_cast_word_raw_pointer2
+// CHECK: unchecked_addr_cast
+// CHECK: } // end sil function 'a2p_p2a_reinterpret_cast_word_raw_pointer2'
+sil [ossa] @a2p_p2a_reinterpret_cast_word_raw_pointer2 : $@convention(thin) (@owned ContiguousArray<UInt16>) -> () {
```
Comment (nate-chandler) [inline test/SILOptimizer/sil_combine_ossa.sil:527]: Does this test currently fail?

**Review by nate-chandler — APPROVED** [review]
LGTM

File: test/SILOptimizer/sil_combine_ossa.sil (hunk 1 above)
Comment (meg-gupta) [reply test/SILOptimizer/sil_combine_ossa.sil:527]: Thanks! I had minimized the test too much.. Modified to a test that will fail without the patch now.

**Comment by meg-gupta** [comment]
@swift-ci smoke test and merge

**Pull request merged**